com.xrtk.spatial-persistence.asa 0.3.0

SimonDarksideJ opened the pull request:
# XRTK - Mixed Reality Toolkit Pull Request

## Overview
<!-- Please provide a clear and concise description of the pull request. -->

Tested implementation ready for release

StephenHodgson committed:
initail project setup

StephenHodgson committed:
Dev/ci cd init (#1)

# XRTK - Mixed Reality Toolkit Pull Request

## Overview
<!-- Please provide a clear and concise description of the pull request. -->
Project setup complete

## Related Submodule Pull Requests

- https://github.com/XRTK/Anchors/pull/1

StephenHodgson committed:
renamed project to XRTK.SpatialPersistence.ASA (#2)

* renamed project to XRTK.SpatialPersistence.ASA

* renamed a few lingering artifacts

* fixed line endings

* updated spatial persistence submodule

SimonDarksideJ committed:
Initial check-in for ASA implementation

SimonDarksideJ committed:
Updated Materials for placed objects

SimonDarksideJ committed:
Updates from Testing

SimonDarksideJ committed:
Updated implementation to be a System instead of an Extension

SimonDarksideJ committed:
Updated implementation based on feedback and updates

SimonDarksideJ committed:
Fixed AnchorLocated Event for ASA Implementation (moved from Place function)

SimonDarksideJ committed:
Updated Spatial Persistence checkout

SimonDarksideJ committed:
Merge pull request #4 from XRTK/feature/asaImplementation

Feature/asa implementation

SimonDarksideJ committed:
Resolved issue with ASA Provider initialisation

SimonDarksideJ committed:
Merge branch 'development' into fix/AnchorInitialisation

SimonDarksideJ committed:
Merge pull request #5 from XRTK/fix/AnchorInitialisation

Resolved issue with ASA Provider initialisation

SimonDarksideJ committed:
Updated implementation from testing

SimonDarksideJ committed:
Updated implementation based on feedback
Fixed Asset Stripping and added Build Process linker to use package link.xml

SimonDarksideJ committed:
Updated implementation with Async options on ASA provider

SimonDarksideJ committed:
Reverting Unity version update due to build server.

StephenHodgson committed:
Added XR Loaders

Updated packages
Removed submodule

StephenHodgson committed:
Fixed build issues

misc cleanup

StephenHodgson committed:
bumped package dependency

StephenHodgson reviewed (changes requested):
Requires https://github.com/XRTK/SpatialPersistence/pull/9 release

StephenHodgson committed:
Added android build info

SimonDarksideJ committed:
Added Android build files

SimonDarksideJ commented:
![](https://media.giphy.com/media/VdiQKDAguhDSi37gn1/giphy.gif?cid=ecf05e47ip12j5gtd3lw06l04h7fbo60rv3monj1cjo975hb&rid=giphy.gif)

SimonDarksideJ commented:
> Is this something we should generate for developers?

No, as it's a custom template for ASA, including references to specific libraries required for ASA to function

StephenHodgson committed:
updated project to 2020.3.17f1 (#7)

* updated project to 2020.3.17f1

fixed a few misc items

* added uwp build

removed broken scene reference

* Fixed uwp builds

* updated dependencies

StephenHodgson committed:
fixed long paths preventing appx signing

SimonDarksideJ committed:
Added Try/Catch to handle obscure errors returned from the ASA system (#8)

# XRTK - Mixed Reality Toolkit Pull Request

## Overview
<!-- Please provide a clear and concise description of the pull request. -->
Added Try/Catch to handle obscure errors returned from the ASA system

## Changes
<!-- Brief list of the targeted features that are being changed. -->
Try/Catch protection added on the result of an Anchor Located event

StephenHodgson committed:
updated packages

SimonDarksideJ committed:
Fixed asset for TryFind and TryDelete

SimonDarksideJ committed:
Merge pull request #9 from XRTK/fix-assert-tryfindanchors

Fixed assert for TryFind and TryDelete

StephenHodgson committed:
GitHub Actions Workflow (#10)

# XRTK - Mixed Reality Toolkit Pull Request

## Overview
Migrate from Azure DevOps to GitHub Actions Workflow

XRTK-Build-Bot committed:
[skip ci] 0.3.0-preview.14

SimonDarksideJ committed:
Updating ASA implementation to allow for infinite anchors (#11)

* Added optional setting of Expiration time only for dates that are the same or in the future (dates in the past will not set an expiration time)
* Updated ASA dependencies
* Updated some Unity package versions

XRTK-Build-Bot committed:
[skip ci] 0.3.0-preview.15

StephenHodgson committed:
removed some misc logging (#12)

a bit of formatting

XRTK-Build-Bot committed:
[skip ci] 0.3.0-preview.16
